src/electron/platform/android/appium-service-configurator-factory.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import ADB from 'appium-adb';
+import {
+    AndroidServiceConfigurator,
+    AndroidServiceConfiguratorFactory,
+} from 'electron/platform/android/android-service-configurator';
+import { AppiumServiceConfigurator } from 'electron/platform/android/appium-service-configurator';
+
+export type AdbCreateParameters = {
+    sdkRoot: string;
+};
+
+export interface AdbCreator {
+    createADB(opts: AdbCreateParameters): Promise<ADB>;
+}
+
+class ProductionAdbCreator implements AdbCreator {
+    public createADB = async (
+        parameters: AdbCreateParameters,
+    ): Promise<AndroidServiceConfigurator> => {
+        return ADB.createADB(parameters);
+    };
+}
+
+export class AppiumServiceConfiguratorFactory implements AndroidServiceConfiguratorFactory {
+    public async getServiceConfigurator(sdkRoot: string): Promise<AndroidServiceConfigurator> {
+        return this.getServiceConfiguratorTestable(sdkRoot, new ProductionAdbCreator());
+    }
+
+    public async getServiceConfiguratorTestable(
```

Comment:
It's not necessary to have a separate "testable" verssion like this; our normal pattern is to leave the responsibility of composing the "real" dependency to the top-level initializer that's composing the dependency tree, and just have most classes implement only functionality that requires injected dependencies. I think the usual pattern would look something like:

```ts
// appium-service-configurator-factory.ts
export type AdbCreator = typeof ADB.createADB;
export class AppiumServiceConfiguratorFactory implements AndroidServiceConfiguratorFactory {
    public constructor(private readonly createAdb: AdbCreator) { }
    // in impl, use "await this.createAdb(parameters)"
}

// unit tests
let mockCreateAdb: IMock<AdbCreator>;
it('invokes createAdb as expected', async () => {
    mockCreateAdb.setup(m => m(expectedParam)).returns(() => Promise.resolve(stubAdbInstance)).verifiable();
    const testSubject = new AppiumServiceConfiguratorFactory(mockCreateAdb);
    // ...
});

// production initializer
const androidServiceConfiguratorFactory = new AppiumServiceConfiguratorFactory(ADB.createADB);
```

Reply:
Let's talk about this. It feels wrong to have the appium-adb dependency leak all the way to the initializer.

Reply:
Discussed offline and agreed on a compromise between these approaches